pkg/ai/amazonsagemaker.go

```diff
@@ -0,0 +1,172 @@
+limitations under the License.
+*/
+
+package ai
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+	"strings"
+
+	"encoding/json"
+	"log"
+
+	"github.com/fatih/color"
+	"github.com/k8sgpt-ai/k8sgpt/pkg/cache"
+	"github.com/k8sgpt-ai/k8sgpt/pkg/util"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/sagemakerruntime"
+)
+
+type SageMakerAIClient struct {
+	client      *sagemakerruntime.SageMakerRuntime
+	language    string
+	model       string
+	temperature float32
+	endpoint    string
+	topP        float32
+	maxTokens   int
+}
+
+type Generations []struct {
+	Generation struct {
+		Role    string `json:"role"`
+		Content string `json:"content"`
+	} `json:"generation"`
+}
+
+type Request struct {
+	Inputs     [][]Message `json:"inputs"`
+	Parameters Parameters  `json:"parameters"`
+}
+
+type Message struct {
+	Role    string `json:"role"`
+	Content string `json:"content"`
+}
+
+type Parameters struct {
+	MaxNewTokens int     `json:"max_new_tokens"`
+	TopP         float64 `json:"top_p"`
+	Temperature  float64 `json:"temperature"`
+}
+
+func (c *SageMakerAIClient) Configure(config IAIConfig, language string) error {
+
+	// Create a new AWS session
+	sess := session.Must(session.NewSessionWithOptions(session.Options{
+		Config:            aws.Config{Region: aws.String(config.GetProviderRegion())},
+		SharedConfigState: session.SharedConfigEnable,
+	}))
+
+	c.language = language
+	// Create a new SageMaker runtime client
+	c.client = sagemakerruntime.New(sess)
+	c.model = config.GetModel()
+	c.endpoint = config.GetEndpointName()
+	c.temperature = config.GetTemperature()
+	c.maxTokens = config.GetMaxTokens()
+	c.topP = config.GetTopP()
+	return nil
+}
+
+func (c *SageMakerAIClient) GetCompletion(ctx context.Context, prompt string, promptTmpl string) (string, error) {
+	// Create a completion request
+
+	if len(promptTmpl) == 0 {
+		promptTmpl = PromptMap["default"]
+	}
+
+	request := Request{
+		Inputs: [][]Message{
+			{
+				{Role: "system", Content: "DEFAULT_PROMPT"},
+				{Role: "user", Content: fmt.Sprintf(promptTmpl, c.language, prompt)},
+			},
+		},
+
+		Parameters: Parameters{
+			MaxNewTokens: int(c.maxTokens),
+			TopP:         float64(c.topP),
+			Temperature:  float64(c.temperature),
+		},
+	}
+
+	// Convert request to []byte
+	bytesData, err := json.Marshal(request)
+	if err != nil {
+		return "", err
+	}
+
+	// Create an input object
+	input := &sagemakerruntime.InvokeEndpointInput{
+		Body:             bytesData,
+		EndpointName:     aws.String(c.endpoint),
+		ContentType:      aws.String("application/json"), // Set the content type as per your model's requirements
+		Accept:           aws.String("application/json"), // Set the accept type as per your model's requirements
+		CustomAttributes: aws.String("accept_eula=true"),
+	}
+
+	// Call the InvokeEndpoint function
+	result, err := c.client.InvokeEndpoint(input)
+	if err != nil {
+		log.Fatal(err)
```

Comment:
This won't block the review, but please remove the fatal here

Reply:
I don't know how I missed that, Thanks!